### PR TITLE
add retry mechanism and apply it and add a way to show extraction progress

### PR DESCRIPTION
**Pull request by A-Harby**
- Add retry mechanism and apply it to the 3 download files.
- Add a way to show extraction progress and ETA.
- Add the needed package in the prerequisites.

### CODE REVIEW

File: apps/prep-env-prereq.sh
Comment (coesensbert) [inline apps/prep-env-prereq.sh:1]: not required as `pv` is installed as one of the prereq's

**Review by coesensbert — COMMENTED** [review]
can't really comment on the code, did you test it? Or should we test it?

**Comment by A-Harby** [comment]
I am testing right now; I will comment once it is done.

image

**Review by coesensbert — APPROVED** [review]
pv install was already there in prereq script, all the rest looks good. Tests went well

**Pull request merged**